src/consensus/aft/raft.h

```diff
@@ -1506,6 +1496,16 @@ namespace aft
       RAFT_TRACE_JSON_OUT(j);
 #endif
 
+      // Ignore if we're not the leader.
+      if (state->leadership_state != kv::LeadershipState::Leader)
+      {
+        RAFT_FAIL_FMT(
+          "Recv append entries response to {} from {}: no longer leader",
+          state->node_id,
+          from);
+        return;
+      }
```

Comment:
Moved this check _after_ the `RAFT_TRACING` log above, so that we go through an appropriate path in trace validation (match a `recv_append_entries_response` line, go through `RcvAppendEntriesResponse`, `DropResponseWhenNotInState`). Without this, we got no line to trace validate this message, but found an alternate compatible behaviour where this message was _dropped_.